frontend/app/src/people/widgetViews/orgnization/WithdrawBudgetModal.tsx

```diff
@@ -0,0 +1,233 @@
+import React, { useState } from 'react';
+import { Wrap } from 'components/form/style';
+import { useIsMobile } from 'hooks/uiHooks';
+import { InvoiceForm, InvoiceInput, InvoiceLabel } from 'people/utils/style';
+import { useStores } from 'store';
+import styled from 'styled-components';
+import { BudgetWithdrawSuccess } from 'store/main';
+import { satToUsd } from 'helpers';
+import LighningDecoder from 'light-bolt11-decoder';
+import { Button, Modal } from '../../../components/common';
+import { colors } from '../../../config/colors';
+import successIcon from '../../../public/static/withdraw_success.svg';
+import errorIcon from '../../../public/static/error.svg';
+import { WithdrawModalProps } from './interface';
+import { Grey } from './style';
+
+const color = colors['light'];
+
+const WithdrawModalTitle = styled.h3`
+  font-size: 1.9rem;
+  font-weight: bolder;
+  margin-bottom: 20px;
+`;
+
+const PaymentDetailsWrap = styled.div`
+  padding: 10px 0px;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+const WithdrawText = styled.p`
+  color: #3c3f41;
+  font-size: 1.05rem;
+  margin-bottom: 30px;
+`;
+
+const WithdrawAmount = styled.h3`
+  font-size: 1.8rem;
+  color: #292c33;
+  padding: 0px;
+  margin: 0px;
+`;
+
+const DollarValue = styled.p`
+  color: #8e969c;
+  font-size: 0.9rem;
+  padding: 0px;
+  margin: 0px;
+`;
+
+const ActionButtonWrap = styled.div`
+  display: flex;
+  margin-top: 40px;
+`;
+
+const SuccessImage = styled.img`
+  width: 56px;
+  height: 56px;
+  margin-bottom: 30px;
+`;
+
+const SuccessText = styled.p`
+  color: #8e969c;
+  font-size: 0.95rem;
+  font-weight: 500;
+  margin-top: 10px;
+`;
+
+const ErrorText = styled(SuccessText)`
+  color: #ed7474;
+`;
+
+const WithdrawBudgetModal = (props: WithdrawModalProps) => {
+  const [paymentRequest, setPaymentRequest] = useState('');
+  const [amountInSats, setAmountInSats] = useState(0);
+  const [paymentSettled, setPaymentSettled] = useState(false);
+  const [paymentError, setPaymentError] = useState('');
+
+  const isMobile = useIsMobile();
+  const { ui, main } = useStores();
+  const { isOpen, close, uuid, getOrganizationBudget } = props;
+
+  function isBudgetSuccess(object: any): object is BudgetWithdrawSuccess {
+    return 'response' in object;
+  }
+
+  const getAndSetDetails = async (paymentRequest: string) => {
```

Comment:
I would move this function into `getInvoiceDetails` since it is not being used in any other scope